x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/authn/SamlAuthnRequestValidator.java

```diff
@@ -264,6 +260,7 @@ private void checkAcs(AuthnRequest request, SamlServiceProvider sp) {
             throw new ElasticsearchSecurityException("The registered ACS URL for this Service Provider is [{}] but the authentication " +
                 "request contained [{}]", sp.getAssertionConsumerService(), acs);
         }
+        authnState.put(SamlAuthenticationState.Fields.ACS_URL.getPreferredName(), acs);
```

Comment:
Is this for template support? Or for concurrency?

Given we know that it _must_ equal the SP's registered ACS, it seems unnecessary here (but I'm not objecting, just trying to understand).

Reply:
As we discussed in person, this is needed so that we can potentially return SAML Responses that indicate failure in the `/init` API, even if we/something fails before we reconstruct the SP object there